### PR TITLE
chore(pt): make `comm_dict` for dpa2 noncompulsory when nghost is 0

### PR DESCRIPTION
**Pull request by njzjz**
Construct and pass a mapping instead when there is no ghost atom.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new test class to evaluate the `DeepPot` model without periodic boundary conditions, enhancing testing coverage.
  
- **Bug Fixes**
	- Corrected the order of returned values in existing tests for clarity and accuracy.

- **Documentation**
	- Updated the cleanup process in test classes to improve test execution consistency. 

- **Chores**
	- Simplified the `TearDown` methods in multiple test classes, removing unnecessary file deletion steps.

### CODE REVIEW

File: source/api_cc/tests/test_deeppot_dpa_pt.cc [hunk 1]
```diff
@@ -163,3 +163,231 @@ TYPED_TEST(TestInferDeepPotDpaPt, cpu_build_nlist_atomic) {
...
+    }
+    for (int ii = 0; ii < natoms; ++ii) {
+      for (int dd = 0; dd < 9; ++dd) {
+        expected_tot_v[dd] += expected_v[ii * 9 + dd];
+      }
+    }
+  };
+
+  void TearDown() override {};
+};
```
[the remark this answers is not in the document]
Comment (njzjz) [reply source/api_cc/tests/test_deeppot_dpa_pt.cc:246]: xref: #3905

**Review by wanghan-iapcm — APPROVED** [review]
